src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.hmcts.reform.blobrouter.tasks.processors;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.specialized.BlobLeaseClient;
+import com.azure.storage.blob.specialized.BlobLeaseClientBuilder;
+import org.slf4j.Logger;
+import uk.gov.hmcts.reform.blobrouter.services.storage.BlobDispatcher;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class BlobProcessor {
+
+    private static final Logger logger = getLogger(BlobProcessor.class);
+
+    private final BlobServiceClient storageClient;
+    private final BlobDispatcher dispatcher;
+
+    public BlobProcessor(
+        BlobServiceClient storageClient,
+        BlobDispatcher dispatcher
+    ) {
+        this.storageClient = storageClient;
+        this.dispatcher = dispatcher;
+    }
+
+    public void process(String blobName, String containerName) {
+        logger.info("Processing {} from {} container", blobName, containerName);
+
+        try {
+            BlobClient blobClient = storageClient
+                .getBlobContainerClient(containerName)
+                .getBlobClient(blobName);
+            BlobLeaseClient leaseClient = new BlobLeaseClientBuilder()
+                .blobClient(blobClient)
+                .buildClient();
+
+            leaseClient.acquireLease(60);
```

Comment:
we should acquire lease only after we have checked blob doesn't exist in database as otherwise it would be difficult for delete job to acquire lease

Reply:
hm. can add db dependency then on next pr. either-way, lease will expire in 60s. and with finally block mentioned below - immediately :)